lib/trino-parquet/src/main/java/io/trino/parquet/reader/ParquetReader.java

```diff
@@ -342,11 +335,9 @@ private void freeCurrentRowGroupBuffers()
         }
 
         for (int column = 0; column < primitiveFields.size(); column++) {
-            Collection<ChunkReader> readers = chunkReaders.get(new ChunkKey(column, currentRowGroup));
-            if (readers != null) {
-                for (ChunkReader reader : readers) {
-                    reader.free();
-                }
+            ChunkedInputStream chunkedStream = chunkReaders.get(new ChunkKey(column, currentRowGroup));
+            if (chunkedStream != null) {
```

Comment:
Immutable map can't contain null entry and `chunkedStream` shouldn't go missing from map here, so we can drop the null check

Reply:
the null check was there before and i think it was because for some row groups, some columns may be missing

Reply:
The null check in previous code had no meaning because multi-map never returns a null, only empty collection.
I don't think it would happen that there is no `chunkedStream` for some `primitiveField` and row group (see constructor), but I'm okay with keeping the null check as well.